tests/test_csc.py

```diff
@@ -304,18 +303,12 @@ async def mount_offset_callback(self):
                     await asyncio.sleep(2.)
 
                 if not get_tel_pos:
-                    await asyncio.sleep(5 * salobj.base_csc.HEARTBEAT_INTERVAL)
                     await harness.aos_remote.cmd_applyCorrection.set_start(azimuth=azimuth,
                                                                            elevation=elevation,
                                                                            timeout=STD_TIMEOUT)
                 else:
```

Comment:
Are you sure it is ok to remove this sleeps? I think the units test may be less reliable without them.

Reply:
I'm a bit confused honestly... because these sleeps used to be 100% necessary until the locks were put in... but you are making the point that the locks aren't technically doing anything as currently implemented, so I'm a little confused.
However, if the locks are working as expected, then the behaviour of the code is such that sleeps should no longer be necessary since commands won't return until completed AND the locks should prevent race conditions.

Reply:
I can assure you the locks are not doing what they where supposed to. You have to acquire a single lock for them to actually work. Adding sleeps like this would be more on the lines of allowing processing time for the underlying coroutines to execute before performing a new task.

Reply:
Sorry, not arguing with you, I just don't quite understand how the sleeps were needed before any locking, yet weren't required when the locks weren't functioning correctly... seems fishy.
Anyways, with the proper lock implementation they still don't appear to be necessary, and they shouldn't be necessary.... so that's probably a good thing.

Reply:
Actually, I bet I know why it worked with the locks having an incorrect implementation. Essentially all methods to apply offsets were waiting for the correction_loop to publish an event saying the correction has been implemented before returning (which still happens). So even though the locks weren't functioning, it wasn't returning early.
That being said, there were probably still ways to get bugs in there with a bit of effort, so it's definitely still better that the locks are implemented properly now.